src/js/util/Topology.js

```diff
@@ -263,12 +263,33 @@ var Topology = function(params, offsets){
 	
         data.links = links;
         
-        if(!data.endpoints) data.endpoints = [];    
+        //Fix for the endpoints bug. 
+        //make endpoints appear on all link endings. 
+
+        if(!data.endpoints) data.endpoints = [];  
+        var tempEndPointsArray = data.endpoints.slice();   
+       data.endpoints = []; 
+       _.forEach(data.links, function(link){
```

Comment:
For a map source that has only endpoints, data.links would be empty and the loop would never execute. Therefore, the nodes would never appear on the map